src/server/graphql/resolvers/Filesystem.js

```diff
@@ -44,17 +44,30 @@ export default {
         },
     },
 
-    FilesystemObject: {
-        __resolveType(obj, _context, _info) {
-            if (obj.type) {
-                if (obj.type.toUpperCase() === "FILE") {
-                    return "File"; // File is the name of the concrete GraphQL type.
-                }
-                if (obj.type === "dir" || obj.type.toUpperCase() === "FOLDER") {
-                    return "Folder"; // Folder is the name of the concrete GraphQL type.
-                }
-            }
-            return null;
+    File: {
+        shareCount: async (file, _args, { dataSources }) => {
+            const stat = await dataSources.terrain.filesystemStat(file.path);
+            return stat["share-count"];
+        },
+    },
+
+    Folder: {
+        fileCount: async (folder, _args, { dataSources }) => {
+            const stat = await dataSources.terrain.filesystemStat(folder.path);
+            return stat["file-count"];
+        },
+        shareCount: async (folder, _args, { dataSources }) => {
+            const stat = await dataSources.terrain.filesystemStat(folder.path);
+            return stat["share-count"];
```

Comment:
Does GraphQL manage caching the result of the call to the `stat` endpoint for us?

Reply:
The caching provided by Apollo doesn't provide batching as well, AFAIK. We're both batching requests and caching it with dataloader. Not sure if the in-process LRU cache would be okay in our environment because of the multiple replicas and I'd really rather not add Redis or memcached to the stack until we have multiple places that absolutely require one of them.

Reply:
Doing my thing and reading about it, I guess Dataloader caches within the context of a single request, but not cross-request. Still a nice improvement, probably, and it's really hard for us to provide nice cache semantics outside of a single request. The caching provided by Apollo might be more useful in non-filesystem contexts where we can provide more useful cache headers in Terrain if we want (of course, it'll also be less useful there, but anyway).